driver/src/test/java/org/neo4j/driver/internal/net/SocketConnectorTest.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver.internal.net;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.neo4j.driver.internal.ConnectionSettings;
+import org.neo4j.driver.internal.security.SecurityPlan;
+import org.neo4j.driver.internal.spi.Connection;
+import org.neo4j.driver.v1.AuthToken;
+import org.neo4j.driver.v1.AuthTokens;
+import org.neo4j.driver.v1.Logging;
+import org.neo4j.driver.v1.exceptions.ClientException;
+
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.RETURNS_MOCKS;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.neo4j.driver.internal.net.BoltServerAddress.LOCAL_DEFAULT;
+
+public class SocketConnectorTest
+{
+    @Test
+    public void connectCreatesConnection()
+    {
+        ConnectionSettings settings = new ConnectionSettings( basicAuthToken() );
+        SocketConnector connector = new RecordingSocketConnector( settings );
+
+        Connection connection = connector.connect( LOCAL_DEFAULT );
+
+        assertThat( connection, instanceOf( ConcurrencyGuardingConnection.class ) );
+    }
+
+    @Test
+    @SuppressWarnings( "unchecked" )
+    public void connectSendsInit()
+    {
+        String userAgent = "agentSmith";
+        ConnectionSettings settings = new ConnectionSettings( basicAuthToken(), userAgent );
+        RecordingSocketConnector connector = new RecordingSocketConnector( settings );
+
+        connector.connect( LOCAL_DEFAULT );
+
+        assertEquals( 1, connector.createConnections.size() );
+        Connection connection = connector.createConnections.get( 0 );
+        verify( connection ).init( eq( userAgent ), any( Map.class ) );
+    }
+
+    @Test
+    public void connectThrowsForUnknownAuthToken()
+    {
+        ConnectionSettings settings = new ConnectionSettings( mock( AuthToken.class ) );
+        RecordingSocketConnector connector = new RecordingSocketConnector( settings );
+
+        try
+        {
+            connector.connect( LOCAL_DEFAULT );
+            fail( "Exception expected" );
+        }
+        catch ( Exception e )
+        {
+            assertThat( e, instanceOf( ClientException.class ) );
+        }
```

Comment:
Yeah, I was thinking to simply copy paste that one assert method into some test-util class if you find it useful. Just FYI though, do as you like =).